src/sardana/pool/poolmeasurementgroup.py

```diff
@@ -123,100 +124,200 @@ def _to_fqdn(name, logger=None):
     return full_name
 
 
-class PoolMeasurementGroup(PoolGroupElement):
+class MeasurementConfiguration(object):
 
     DFT_DESC = 'General purpose measurement group'
 
-    def __init__(self, **kwargs):
-        self._state_lock = threading.Lock()
-        self._monitor_count = None
-        self._nr_of_starts = 1
-        self._pending_starts = 0
-        self._acquisition_mode = AcqMode.Timer
+    def __init__(self, parent=None):
+        self._parent = None
+        if parent is not None:
+            self._parent = weakref.ref(parent)()
```

Comment:
I think tha this way you are creating a real reference and not a weakref.

Reply:
Yes, you're right, I'll change for the proxy: https://docs.python.org/2.7/library/weakref.html

> weakref.proxy(object[, callback])
    Return a proxy to object which uses a weak reference. This supports use of the proxy in most contexts instead of requiring the explicit dereferencing used with weak reference objects. The returned object will have a type of either ProxyType or CallableProxyType, depending on whether object is callable. Proxy objects are not hashable regardless of the referent; this avoids a number of problems related to their fundamentally mutable nature, and prevent their use as dictionary keys. callback is the same as the parameter of the same name to the ref() function.